web/modules/custom/joinup_discussion/src/EventSubscriber/SubscribedDiscussionCommentSubscriber.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace Drupal\joinup_discussion\EventSubscriber;
+
+use Drupal\joinup_notification\Event\NotificationEvent;
+use Drupal\joinup_notification\JoinupMessageDeliveryInterface;
+use Drupal\joinup_notification\NotificationEvents;
+use Drupal\joinup_subscription\JoinupSubscriptionInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Subscribes to comment CRUD operations.
+ */
+class SubscribedDiscussionCommentSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The Joinup subscribe service.
+   *
+   * @var \Drupal\joinup_subscription\JoinupSubscriptionInterface
+   */
+  protected $subscribeService;
+
+  /**
+   * The Joinup message delivery service.
+   *
+   * @var \Drupal\joinup_notification\JoinupMessageDeliveryInterface
+   */
+  protected $messageDelivery;
+
+  /**
+   * The comment entity.
+   *
+   * @var \Drupal\comment\CommentInterface
+   */
+  protected $comment;
+
+  /**
+   * The comment parent discussion node.
+   *
+   * @var \Drupal\node\NodeInterface
+   */
+  protected $discussion;
+
+  /**
+   * The comment parent discussion group.
+   *
+   * @var \Drupal\rdf_entity\RdfInterface
+   */
+  protected $group;
+
+  /**
+   * The list of recipients as user accounts.
+   *
+   * @var \Drupal\user\UserInterface[]
+   */
+  protected $recipients;
+
+  /**
+   * Constructs a new event subscriber object.
+   *
+   * @param \Drupal\joinup_subscription\JoinupSubscriptionInterface $subscribe_service
+   *   The Joinup subscribe service.
+   * @param \Drupal\joinup_notification\JoinupMessageDeliveryInterface $message_delivery
+   *   The Joinup message delivery service.
+   */
+  public function __construct(JoinupSubscriptionInterface $subscribe_service, JoinupMessageDeliveryInterface $message_delivery) {
+    $this->subscribeService = $subscribe_service;
+    $this->messageDelivery = $message_delivery;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents(): array {
+    return [NotificationEvents::COMMENT_CRUD => 'commentCrudProxy'];
+  }
+
+  /**
+   * Acts as a proxy by passing the control to a dedicated method.
+   *
+   * Normally we should have split the events into more specific events, like
+   * 'comment create', 'comment update', 'comment delete'. Just 'comment crud'
+   * is too generic and is not actually an event. In order to keep the
+   * compatibility with Joinup Notification module, we use this proxy method to
```

Comment:
I agree fully with this, maybe we should make a ticket to refactor this in the future so it uses distinct events?